addons/binding/org.openhab.binding.chromecast/src/main/java/org/openhab/binding/chromecast/handler/ChromecastHandler.java

```diff
@@ -342,36 +425,74 @@ private void handleCcMediaStatus(final MediaStatus mediaStatus) {
+                new DecimalType(mediaStatus.currentTime));
+
+        updateMediaInfoStatus(mediaStatus.media);
     }
 
-    private void handleCcVolume(final Volume volume) {
-        PercentType value = new PercentType((int) (volume.level * 100));
-        updateState(new ChannelUID(getThing().getUID(), ChromecastBindingConstants.CHANNEL_VOLUME), value);
-        this.volume = value;
-        updateState(new ChannelUID(getThing().getUID(), ChromecastBindingConstants.CHANNEL_MUTE),
-                volume.muted ? OnOffType.ON : OnOffType.OFF);
+    private void updateMediaInfoStatus(final Media media) {
+        Double duration = 0.0;
+        String metadataType = Media.MetadataType.GENERIC.name();
+
+        if (media != null) {
+            duration = media.duration;
+            metadataType = media.getMetadataType().name();
+        }
+
+        updateState(new ChannelUID(getThing().getUID(), ChromecastBindingConstants.CHANNEL_DURATION),
+                duration == null ? UnDefType.UNDEF : new DecimalType(duration));
+        updateState(new ChannelUID(getThing().getUID(), ChromecastBindingConstants.CHANNEL_METADATA_TYPE),
+                new StringType(metadataType));
+
+        updateMetadataStatus(media == null ? Collections.emptyMap() : media.metadata);
     }
 
-    @Override
-    public void spontaneousEventReceived(final ChromeCastSpontaneousEvent event) {
-        switch (event.getType()) {
-            case MEDIA_STATUS:
-                final MediaStatus mediaStatus = event.getData(MediaStatus.class);
-                handleCcMediaStatus(mediaStatus);
-                break;
-            case STATUS:
-                final Status status = event.getData(Status.class);
-                handleCcStatus(status);
-                break;
-            case UNKNOWN:
-                logger.debug("Received an 'UNKNOWN' event (class={})", event.getType().getDataClass());
-                break;
-            default:
-                logger.debug("Unhandled event type: {}", event.getData());
-                break;
+    private void updateMetadataStatus(Map<String, Object> metadata) {
+        getThing().getChannels().stream() //
+                .map(channel -> channel.getChannelTypeUID().getId())
+                .filter(channel -> ArrayUtils.contains(ChromecastBindingConstants.METADATA_CHANNELS, channel))
+                .forEach(channel -> updateChannel(channel, metadata));
+    }
+
+    private void updateChannel(String channelId, Map<String, Object> metadata) {
+        if (!isLinked(channelId)) {
+            return;
+        }
 
+        Object value = getValue(channelId, metadata);
+        State state = UnDefType.UNDEF;
+
+        if (value == null) {
+            state = UnDefType.UNDEF;
+        } else if (value instanceof Double) {
+            state = new DecimalType((Double) value);
+        } else if (value instanceof Integer) {
+            state = new DecimalType(BigDecimal.valueOf(((Integer) value).longValue()));
```

Comment:
You can also request the longValue of an Integer you don't need this indirection https://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html#longValue()